---
title: "eClip-seq_Analysis"
author: "Pankti"
date: "11/9/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Loading the libraries:
```{r, message=F, warning=F, echo=FALSE}
library(org.Hs.eg.db)
library(clusterProfiler)
library(rtracklayer)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(ReactomePA)
library(ggvenn)
library(ChIPseeker)
library(GenomicFeatures)
```


```{r}
file <- "/shared/data/results/pgosar/clipseq/analysis/hepg2_Lin28b.8nt.peaks.bed"
# ref_file <- "/shared/nextrna-resources/grch38/gencode.v42.annotation.sorted.gtf" # Not using this
bed <- rtracklayer::import(file)
```


```{r}
# bed_df <- read.table(file)
```


```{r}

```

ChIPseeker
```{r, fig.align='center', fig.width=20, fig.height=20}
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
peak <- readPeakFile(file)
# peak
covplot(peak, weightCol = "V5", title = "eClip Peaks over Chromosomes") #works
peakHeatmap(file, TxDb = txdb, upstream = 3000, downstream = 3000, color="red") #works Not very significant though
```

# Profile of clip peaks binding to TSS regions
```{r}
promoter <- getPromoters(TxDb = txdb, upstream=3000, downstream=3000)
TagMatrix <- getTagMatrix(peak, windows = promoter)
plotAvgProf2(file, TxDb = txdb, xlab = "Genomic Region (5'->3')", ylab = "Read count frequency", upstream=3000, downstream=3000, conf=0.95) #works
```
# Not working
```{r}
# plotPeakProf2(peak, upstream = rel(0.2), downstream = rel(0.2), conf = 0.95,
#               by="gene", type = "body", nbin = 800, TxDb = txdb,
#               weightCol = "V5", ignore_strand = F)
```

# Annontating the file
TSS is transcription start site
``` {r}
seqlevelsStyle(txdb) <- "UCSC"
peakAnno <- annotatePeak(file, tssRegion=c(-3000, 3000), TxDb=txdb, annoDb="org.Hs.eg.db")
plotAnnoPie(peakAnno)
plotAnnoBar(peakAnno)
vennpie(peakAnno)
upsetplot(peakAnno)
upsetplot(peakAnno, vennpie=T)
```


``` {r}
# library(ReactomePA)
pathway1 <- enrichPathway(as.data.frame(peakAnno)$geneId)
head(pathway1, 2)
```


``` {r, fig.height=8}
gene <- seq2gene(peak, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb = txdb)
pathway2 <- enrichPathway(gene)
head(pathway2, 2)
dotplot(pathway2)
```
# Gene Set Enrichment Analysis
The gene ratio is the ratio of the enriched genes with respect to the total number of genes in the pathway
# Kegg Pathway
``` {r, fig.height=8}
peakAnno_kegg <- enrichKEGG(as.data.frame(peakAnno)$geneId, pvalueCutoff = 0.05, pAdjustMethod = "BH")
dotplot(peakAnno_kegg, showCategory= 20, title="KEGG Pathway Enrichment Analysis")
```

# Biological Processes
``` {r, fig.height=10}
enrichgobp <- enrichGO(gene, 'org.Hs.eg.db', ont="BP", pvalueCutoff = 0.05, pAdjustMethod = "BH", readable = T)
dotplot(enrichgobp, showCategory= 20, title="GO biological process Pathway Enrichment Analysis")
```


``` {r, fig.height=10}
enrichgoMF <- enrichGO(gene, 'org.Hs.eg.db', ont="MF", pvalueCutoff = 0.05, pAdjustMethod = "BH", readable = T)
dotplot(enrichgoMF, showCategory= 20, title="GO Molecular Functions Pathway Enrichment Analysis")
```


``` {r}
peakHeatmap(peak, weightCol = "V5", TxDb = txdb, upstream = 1000, downstream = 1000,
  title = "Heatmap", verbose = TRUE, color = "midnightblue")
```


``` {r}

```


``` {r}

```


``` {r}

```
